homeassistant/components/switchbee/cover.py

```diff
@@ -23,59 +23,91 @@
 from .coordinator import SwitchBeeCoordinator
 from .entity import SwitchBeeDeviceEntity
 
-_DeviceTypeT = TypeVar(
-    "_DeviceTypeT",
-    bound=Union[SwitchBeeShutter, SwitchBeeSomfy],
-)
-
 
 async def async_setup_entry(
     hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
 ) -> None:
     """Set up SwitchBee switch."""
     coordinator: SwitchBeeCoordinator = hass.data[DOMAIN][entry.entry_id]
     async_add_entities(
-        SwitchBeeCover(device, coordinator)
+        SwitchBeeCoverEntity(device, coordinator)
         for device in coordinator.data.values()
         if isinstance(
             device,
-            (
-                SwitchBeeShutter,
-                SwitchBeeSomfy,
-            ),
+            SwitchBeeShutter,
         )
     )
 
+    async_add_entities(
```

Comment:
I think you shouldn't call `async_add_entities` twice. There should be a way to use `itertools.chain` for this

Reply:
I can do this:

```python
async def async_setup_entry(
    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
) -> None:
    """Set up SwitchBee switch."""
    coordinator: SwitchBeeCoordinator = hass.data[DOMAIN][entry.entry_id]
    switchbee_covers_tuple = []
    for device in coordinator.data.values():
        if isinstance(device, SwitchBeeShutter):
            switchbee_covers_tuple.append((device, SwitchBeeCoverEntity))
        elif isinstance(device, SwitchBeeSomfy):
            switchbee_covers_tuple.append((device, SwitchBeeSomfyEntity))

    async_add_entities(
        device_class(device, coordinator)
        for device, device_class in switchbee_covers_tuple
    )
```